### PR TITLE
Update visualization chooser list when new entries are available

### PR DESCRIPTION
**Pull request by farmaazon**
### Pull Request Description
The visualization chooser, once created, never get its list updated when new visualizations were read from the project directory. Now it refreshes its list when it's brought to screen.

Also fixed leak in drop-down component on this occasion.

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Rust](https://github.com/luna/enso/blob/main/docs/style-guide/rust.md) style guide.
- [x] All code has been tested where possible.
- [x] All code has been profiled where possible.

### CODE REVIEW

File: src/rust/ide/view/graph-editor/src/component/visualization/container/visualization_chooser.rs [hunk 1]
```diff
@@ -100,23 +106,31 @@ impl VisualizationChooser {
...
+            eval  frp.set_menu_offset_y ((offset) menu.set_menu_offset_y.emit(offset) );
+
+            _set_selected <- frp.input.set_selected.map2(&frp.output.entries,
+                f!([menu](selected,entries) {
+                    if let Some(selected) = selected {
+                        let ix = entries.iter().position(|item| item == selected);
+                        menu.set_selected.emit(ix);
+                    }
+                })
+            );
```
Comment (wdanilo) [inline src/rust/ide/view/graph-editor/src/component/visualization/container/visualization_chooser.rs:121]: Could you write it more functional than imperative? I mean something along the lines of:

```rust
            ix <= frp.input.set_selected.map2(&frp.output.entries,
                f!([menu](selected,entries) selected.map(|s| entries.iter().position(|item| item == selected)))
            );
            eval ix ((ix) menu.set_selected.emit(ix));
```

Btw the last line should be `menu.set_selected <+ ix` in FRP3.

The rule of thumb for me is that if we can "emit" things as the ONLY operation in eval / map, we should do it this way.

File: src/rust/ide/view/graph-editor/src/component/visualization/container/visualization_chooser.rs [hunk 2]
```diff
@@ -100,23 +106,31 @@ impl VisualizationChooser {
...
+            // === Showing Entries ===
+
 
-            eval frp.set_icon_size ((size) menu.set_icon_size.emit(size) );
-            eval frp.set_icon_padding ((size) menu.set_icon_padding.emit(size) );
-            eval frp.hide_selection_menu ((size) menu.hide_selection_menu.emit(size) );
-            eval frp.set_menu_offset_y ((offset) menu.set_menu_offset_y.emit(offset) );
+            frp.source.entries <+ menu.menu_visible.gate(&menu.menu_visible).map(f_!([model] {
+                let entries                                     = Rc::new(model.entries());
+                let provider:list_view::entry::AnyModelProvider = entries.clone_ref().into();
```
Comment (wdanilo) [inline src/rust/ide/view/graph-editor/src/component/visualization/container/visualization_chooser.rs:129]: much nicer layout of code would be if you'd write it as `let provider = list_view::entry::AnyModelProvider::from(entries.clone_ref());`

File: src/rust/ide/view/graph-editor/src/component/visualization/container/visualization_chooser.rs [hunk 3]
```diff
@@ -125,11 +139,13 @@ impl VisualizationChooser {
...
-            selected_path <- model.selection_menu.frp.chosen_entry.map(f!([model](entry_id)({
-                entry_id.map(|entry_id| {
-                    model.visualization_alternatives.borrow().get(entry_id).cloned()
-                }).flatten()
-            })));
+            selected_path <- model.selection_menu.frp.chosen_entry.map2(&frp.output.entries,
+                f!([](entry_id,entries) {
+                    entry_id.map(|entry_id| {
+                        entries.get(entry_id).cloned()
+                    }).flatten()
```
Comment (wdanilo) [inline src/rust/ide/view/graph-editor/src/component/visualization/container/visualization_chooser.rs:146]: why not oneliner here without curly braces?

**Review by wdanilo — APPROVED** [review]

**Comment by MichaelMauderer** [comment]
Looks good to me! One general idea: should the visualization registry at some point have its own FRP API that can update about changes to the available visualizations? @wdanilo and I were discussing something like this when thinking about theme colors: having a function that returns an FRP endpoint. In this case, for example, `fn valid_sources_frp([...]) -> frp::Stream<Vec<visualization::Definition>>`

**Review by MichaelMauderer — APPROVED** [review]

**Pull request merged**